13.go

```diff
@@ -1032,12 +1042,23 @@ func (hs *clientHandshakeState) doTLS13Handshake() error {
 		return unexpectedMessageError(certMsg, msg)
 	}
 	hs.keySchedule.write(certMsg.marshal())
+
 	// Validate certificates.
 	certs := getCertsFromEntries(certMsg.certificates)
 	if err := hs.processCertsFromServer(certs); err != nil {
 		return err
 	}
 
+	// Validate the DC if present. The DC is only processed if the extension was
+	// indicated by the ClientHello; otherwise this call will result in an
+	// "illegal_parameter" alert.
+	if len(certMsg.certificates) > 0 {
+		if err := hs.processDelegatedCredentialFromServer(
+			certMsg.certificates[0].delegatedCredential); err != nil {
+			return err
+		}
+	}
+
```

Comment:
I think there is a bug in case TLSv1.3 is used.

Chapter 3 of the spaec says:
```
   On receiving a credential and a certificate chain, the client
   validates the certificate chain and matches the end-entity
   certificate to the server's expected identity following its normal
   procedures.  It then takes the following additional steps:
```
This part is done indeed, but then it says:

```
   o  Use the public key in the credential to verify a signature
      provided in the handshake.  That is the CertificateVerify message
      in TLS 1.3 or ...

```
I don't think it is done. In order to do that, you would need to modify call to ``verifyPeerCertificate()``. Currently call to this function takes ``hs.c.peerCertificate[0].PublicKey``, which is not a delegated credential key (except if I'm missing something).

BTW:  Better to merge #104 before (name of this function is missleading)

Reply:
Yup, for TLS 1.3 it looks like I forgot to change the key used to sign/verify the handshake when using a DC. The next commit fixes this and adds a test case. Thanks for catching this.